apps/website/auto-api.ts

```diff
@@ -63,93 +64,99 @@ type ParsedProps = {
+      if (capture.name === 'prop') {
+        parsedProps.prop = capture.node.text;
       }
-
-      if (lol.name === 'type') {
-        parsedProps.type = lol.node.text;
-        if (subComponentName === topKey(last)) {
-          last[topKey(last)].push(parsedProps);
+      if (capture.name === 'type') {
+        parsedProps.type = capture.node.text;
+        const lastKey = last ? Object.keys(last)[0] : '';
+        if (subComponentName === lastKey) {
+          last![lastKey].push(parsedProps);
         } else {
           parsed.push({ [subComponentName]: [parsedProps] });
         }
       }
     });
   });
 
-  const completeSubComponent: SubComponent = { [component_name[1]]: parsed };
+  const completeSubComponent: SubComponent = { [componentName]: parsed };
   ref.push(completeSubComponent);
   return ref;
 }
 
 function writeToDocs(fullPath: string, componentName: string, api: ComponentParts) {
   if (fullPath.includes('kit-headless')) {
-    const relDocPath = `../website/src/routes//docs/headless/${componentName}`;
+    const relDocPath = `../website/src/routes/docs/headless/${componentName}`;
     const fullDocPath = resolve(__dirname, relDocPath);
-    const dirPath = fullDocPath.concat('/auto-api');
+    const dirPath = resolve(fullDocPath, 'auto-api');
 
     if (!fs.existsSync(dirPath)) {
-      fs.mkdirSync(dirPath);
+      fs.mkdirSync(dirPath, { recursive: true });
     }
     const json = JSON.stringify(api, null, 2);
-    const hacky = `export const api=${json}`;
+    const exportedApi = `export const api = ${json};`;
 
     try {
-      fs.writeFileSync(dirPath.concat('/api.ts'), hacky);
-      console.log('auto-api: succesfully genereated new json!!! :)');
+      fs.writeFileSync(resolve(dirPath, 'api.ts'), exportedApi);
+      console.log('auto-api: successfully generated new JSON!');
     } catch (err) {
-      return;
+      console.error('Error writing API file:', err);
     }
   }
 }
+
 function loopOnAllChildFiles(filePath: string) {
-  const childComponentRegex = /\/([\w-]*).tsx$/.exec(filePath);
-  if (childComponentRegex === null) {
+  const childComponentMatch = /[\\/](\w[\w-]*)\.tsx$/.exec(filePath);
+  if (!childComponentMatch) {
     return;
   }
-  const parentDir = filePath.replace(childComponentRegex[0], '');
-  const componentRegex = /\/(\w*)$/.exec(parentDir);
-  if (!fs.existsSync(parentDir) || componentRegex == null) {
+  const parentDir = filePath.slice(0, filePath.lastIndexOf(childComponentMatch[0]));
+  const componentMatch = /[\\/](\w+)$/.exec(parentDir);
+  if (!fs.existsSync(parentDir) || !componentMatch) {
     return;
   }
-  const componentName = componentRegex[1];
+  const componentName = componentMatch[1];
   const allParts: SubComponents = [];
   const store: ComponentParts = { [componentName]: allParts };
+
   fs.readdirSync(parentDir).forEach((fileName) => {
-    if (/tsx$/.test(fileName)) {
-      const fullPath = parentDir + '/' + fileName;
+    if (/\.tsx$/.test(fileName)) {
```

Comment:
Another good catch. While it would take some adventurous file naming, the previous regex could give false positives